specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,387 @@
+                                MB_OK);
+                            MessageBox(nullptr, stringData.get(), L"ExecuteScript Result",
+                                MB_OK);
+                        }
+                    }
+                }
+                return S_OK;
+            })
+            .Get());
+}
+
+
+// The second case is when the execution throws an exception due to a regex error,
+// we can catch and handle it via `get_Exception` in c++.
+void MatchRegWithScript_2(wil::com_ptr<ICoreWebView2> m_webView)
+{
+    wil::com_ptr<ICoreWebView2Staging2> webview2 = m_webView.try_query<ICoreWebView2Staging2>();
+
+    auto scriptCode = GenerateScriptCode(L"This is a string", L"/a", L"It");
+    webview2->ExecuteScriptWithResult(
+        scriptCode.c_str(),
+        Callback<ICoreWebView2StagingExecuteScriptWithResultCompletedHandler>(
+            [](
+                HRESULT errorCode, ICoreWebView2StagingExecuteScriptResult* result) -> HRESULT
+            {
+                if (errorCode != S_OK || result == nullptr)
+                {
+                    MessageBox(
+                        nullptr, L"Get execute status failed!", L"ExecuteScript Result", MB_OK);
+                }
+                else
+                {
+                    wil::com_ptr<ICoreWebView2StagingExecuteScriptException> exception;
+                    BOOL is_success;
+                    // The is_success will return false since an incorrect regular expression
+                    // was passed in.
+                    if (result->get_IsSuccess(&is_success) == S_OK && !is_success)
+                    {
+                        result->get_Exception(&exception);
+
+                        // Get the exception name, this could return the empty string,
+                        // such as `throw 1`.
+                        // In this case, we will get 'SyntaxError'.
+                        wil::unique_cotaskmem_string exceptionName;
+                        if (exception && exception->get_Name(&exceptionName) == S_OK)
+                        {
+                            MessageBox(
+                                nullptr, exceptionName.get(), L"ExecuteScript Result", MB_OK);
+                        }
+
+                        // Get the exception message, this could return the empty
+                        // string, such as `throw 1`.
+                        // In this case, we will get 'SyntaxError: Invalid regular expression:
+                        // missing /'.
+                        wil::unique_cotaskmem_string exceptionMessage;
+                        if (exception && exception->get_Message(&exceptionMessage) == S_OK)
+                        {
+                            MessageBox(
+                                nullptr, exceptionMessage.get(), L"ExecuteScript Result",
+                                MB_OK);
+                        }
+
+                        // Get the exception detail, it's a json struct data with all
```

Comment:
The sample doesn't show how to parse ExceptionAsJSON though it just shows it to the end user. This is probably unrealistic and in actual code we would expect the end dev's code to parse the ExceptionAsJSON string. But for this sample (and probably most end dev's code) it is not necessary to parse it. Instead I would recommend removing this get_ExceptionAsJSON since the sample doesn't use it.